xbmc/Application.cpp

```diff
@@ -3568,6 +3572,8 @@ void CApplication::OnPlayBackStopped()
   CDarwinUtils::EnableOSScreenSaver(true);
 #endif
 
+  CServiceBroker::GetPVRManager().OnPlaybackStopped(m_itemCurrentFile);
+
```

Comment:
ApplicationMessenger is for messages in direction to the main thread.
This code is good as is. It is supposed to be snychronous, not to be hooked in to some undefined messaging anarchy